lib/tessel/name.js

```diff
@@ -102,6 +102,11 @@ Tessel.prototype.rename = function(opts, callback) {
   else {
     self.getName(function(err, oldName) {
       if (err) return callback && callback(err);
+      // Don't set the new name if it's the same as the old name
+      if (oldName == opts.newName) {
+        tessel.logs.info("Name of device is already", oldName);
```

Comment:
Yes I considered that. I don't think it's an error though. Happy to change it if you feel strongly.